autonat.go

```diff
@@ -297,7 +295,7 @@ func (as *AmbientAutoNAT) recordObservation(observation autoNATResult) {
 		as.status.Store(autoNATResult{network.ReachabilityUnknown, nil})
 		if currentStatus.Reachability != network.ReachabilityUnknown {
 			if as.service != nil {
-				as.service.Disable()
+				as.service.Enable()
```

Comment:
I'm not sure if we should flip back to enabled if we _were_ private. But I guess this is symmetric. Up to you.

Reply:
the simpler state machine seems nicer. i'll stick with this for now.